sdk/attestation/azure-security-attestation/samples/async_samples/sample_attest_enclave_async.py

```diff
@@ -0,0 +1,248 @@
+        oe_report = base64.urlsafe_b64decode(sample_open_enclave_report)
+        runtime_data = base64.urlsafe_b64decode(sample_runtime_data)
+        print()
+        print('Attest Open enclave using ', self.shared_url)
+
+        # [START attest_open_enclave_shared_with_options]
+
+        def validate_token(token, signer):
+            #type(AttestationToken, AttestationSigner) -> bool
+            """
+        Perform minimal validation of the issued SGX token.
+        The token validation logic will have checked the issuance_time
+        and expiration_time, but this shows accessing those fields.
+        
+        The validation logic also checks the subject of the certificate to verify
+        that the issuer of the certificate is the expected instance of the service.
+        """
+            print("In validation callback, checking token...")
+            print("     Token issuer: ", token.issuer)
+            print("     Token was issued at: ", token.issuance_time)
+            print("     Token expires at: ", token.expiration_time)
+            if token.issuer != self.shared_url:
+                print("Token issuer {} does not match expected issuer {}".format(token.issuer, self.shared_url))
+                return False
+
+            # Check the subject of the signing certificate used to validate the token.
+            certificate = cryptography.x509.load_der_x509_certificate(signer.certificates[0], backend=default_backend())
+            if certificate.subject != x509.Name([x509.NameAttribute(NameOID.COMMON_NAME, self.shared_url)]):
+                print("Certificate subject {} does not match expected subject {}".format(certificate.subject, self.shared_url))
+                return False
+
+            print("Token passes validation checks.")
+            return True
+
+        async with self._create_client(self.shared_url,
+            token_validation_options=TokenValidationOptions(
+                validation_callback=validate_token)) as attest_client:
+            response = await attest_client.attest_open_enclave(
+                oe_report, runtime_data=AttestationData(runtime_data, is_json=False))
+
+        print("Issuer of token is: ", response.value.issuer)
+        # [END attest_open_enclave_shared_with_options]
+
+    def _create_client(self, base_url, **kwargs):
+        #type:(str, Dict[str, Any]) -> AttestationClient
+        tenant_id = os.getenv("ATTESTATION_TENANT_ID")
+        client_id = os.getenv("ATTESTATION_CLIENT_ID")
+        secret = os.getenv("ATTESTATION_CLIENT_SECRET")
+
+        if tenant_id and client_id and secret:
+            # Create azure-identity class
+            from azure.identity.aio import ClientSecretCredential
+
+            credentials = ClientSecretCredential(
+                tenant_id=tenant_id, client_id=client_id, client_secret=secret
+            )
+
+        return AttestationClient(credentials, instance_url=base_url, **kwargs)
```

Comment:
```suggestion
        self.client = AttestationClient(credentials, instance_url=base_url, **kwargs)
        yield
        self.client.close() # Only do this if you're not using the client in a context manager
        await credentials.close()
```
This should take care of the unclosed sessions you were seeing. You'll have to change your code a bit because we can't return an object from a `yield` line.